### ISSUE TITLE
Suggestion: ChatCommand "/delete" shall not delete attached grids

### ISSUE THREAD

**Issue by Spcemarine** [SEModCommunity/SE-Community-Mod-API#126]
When I perform the command "/delete [all, station, ship] nobeacon" all grids without a beacon are deleted. So far not a bug but I would like to suggest to spare the grids that are attached on pistons and rotors if the "main grid" has a beacon. We would like to delete some trash but if we do it stuff like hangar gates will also disappear.
BTW: You are doing great stuff!

**Comment by mikeloeven** [comment]
yes seconded i was cleaning my server and deleted a few unknown grids only to find out that they were the pistons on my drilling rig XD not fun to replace. though this is more keens fault for not giving sub-grids better identification and i am not sure if the script can even tell what is and is not attached

**Comment by Spcemarine** [comment]
Well, I browsed through the code and couldn't find a way to detect whether a grid is attached or not, otherwise I would do it on my own. Since the API is not finished yet it may be possible for the creator to find a way. 
I don't think it's anyone's fault because keen is not involved in the developement of this API and the API is still in developement.

**Comment by chessmaster42** [comment]
Yeah this is doable but it's no easy task. I'm slowly getting the bits of the API built out to support rotors and pistons and once that's done those blocks will have a property that indicates any attached cube grids.

Until that point it's not possible to do what you're asking but it is something we're working on.

**Comment by Spcemarine** [comment]
Ok, very well. So it's like I thought. Will you change the "/delete" command when you finished it or am I supposed to implement my own commands?

**Comment by chessmaster42** [comment]
Yes, the default command will be updated once implementation of all of MergeBlockEntity, PistonEntity, and RotorEntity is complete. Right now the first two are finished but RotorEntity is still being worked on. So changes are good that you will see the change in the next dev build or two.

**Comment by Spcemarine** [comment]
That's great! Thank you very much.

**Comment by fvgoulet** [comment]
@Spcemarine Is this issue fixed?
